q2_feature_table/tests/test_filter.py

```diff
@@ -340,6 +437,252 @@ def test_combine_frequency_filters(self):
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=4)
+        expected = Table(np.array([[1], [1]]),
+                         ['O1', 'O2'],
+                         ['S2'])
+        self.assertEqual(actual, expected)
+
+        # exclude one, max & min_frequency filter none
+        df = pd.DataFrame({'Subject': ['subject-1'],
+                           'SampleType': ['gut']},
+                          index=['S1'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=42,
+                                min_frequency=0)
+        expected = Table(np.array([[1, 3], [1, 2]]),
+                         ['O1', 'O2'],
+                         ['S2', 'S3'])
+        self.assertEqual(actual, expected)
+
+        # exclude one, min_frequency filter one,
+        # max_frequency filter one
+        df = pd.DataFrame({'Subject': ['subject-1'],
+                           'SampleType': ['gut']},
+                          index=['S1'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                max_frequency=4,
+                                min_frequency=3)
+        expected = Table(np.array([]), [], [])
+        self.assertEqual(actual, expected)
+
+        # where filter one -> exclude one,
+        # min_frequency filter one,
+        # max_frequency filter one
+        df = pd.DataFrame({'Subject': ['subject-1', 'subject-2'],
+                           'SampleType': ['gut', 'tongue']},
+                          index=['S1', 'S2'])
+        metadata = qiime2.Metadata(df)
+        table = Table(np.array([[0, 1, 3], [1, 1, 2]]),
+                      ['O1', 'O2'],
+                      ['S1', 'S2', 'S3'])
+        where = "Subject='subject-1'"
+        actual = filter_samples(table,
+                                metadata=metadata,
+                                exclude_ids=True,
+                                where=where,
+                                max_frequency=4,
+                                min_frequency=3)
+        expected = Table(np.array([]), [], [])
+        self.assertEqual(actual, expected)
+
```

Comment:
Also, a few more tests in this section:

- frequency filtering:
  - exclude one by id, filter same sample by min filter (variation of exclude one, min_frequency filter one)
  - exclude one by id, filter sample sample by max filter
- feature filtering:
  - exclude one by id, filter same sample by min filter (variation of exclude one, min_frequency filter one)
  - exclude one by id, filter sample sample by max filter